scripts/update_robots.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+
+#------------------------------------------------------------------------------
+# Prints version of the Solidity compiler that the source code corresponds to.
+#
+# The documentation for solidity is hosted at:
+#
+#     https://docs.soliditylang.org
+#
+# ------------------------------------------------------------------------------
+# This file is part of solidity.
+#
+# solidity is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# solidity is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with solidity.  If not, see <http://www.gnu.org/licenses/>
+#
+# (c) 2017 solidity contributors.
+#------------------------------------------------------------------------------
+
+set -e
+set -u
+
+PATH_TO_GET_VERSION_SCRIPT="./scripts/get_version.sh"
+PATH_TO_ROBOTS_TXT_FILE="docs/_static/robots.txt"
+TEMP_FILE="temp_store_robots.txt"
+VERSION_FROM_SCRIPT=`bash ${PATH_TO_GET_VERSION_SCRIPT} | perl -pe 'chomp'`
+LAST_LINE_ROBOTS=`tail -n 1 ${PATH_TO_ROBOTS_TXT_FILE}`
+VERSION_NUMBER_IN_ROBOTS="#\w+:\s\/([a-z]+)\/v([0-9]\.[0-9]\.[0-9])\/"
+[[ ${LAST_LINE_ROBOTS} =~ ${VERSION_NUMBER_IN_ROBOTS} ]]
+if [[ $? == 0 ]]; then
+    if [[ ${BASH_REMATCH[2]} != ${VERSION_FROM_SCRIPT} ]]; then
+        sed '$d' ${PATH_TO_ROBOTS_TXT_FILE} > ${TEMP_FILE}; cat ${TEMP_FILE} > ${PATH_TO_ROBOTS_TXT_FILE}
+        echo "Disallow: /${BASH_REMATCH[1]}/v${BASH_REMATCH[2]}/" >> ${PATH_TO_ROBOTS_TXT_FILE}
+        echo "#Disallow: /${BASH_REMATCH[1]}/v${VERSION_FROM_SCRIPT}/" >> ${PATH_TO_ROBOTS_TXT_FILE}
```

Comment:
The script makes an assumption that the line we want to change is the last one in the file. I don't think this is a good assumption. We're likely to add other things to the file later and the script should handle them gracefully. It would be best if we don't have to update this script every time someone changes something in `robots.txt`.

The current structure of the file is not well suited for detecting the right `Disallow` line to change so my proposal would be to change its structure a bit. First let's manually update `robots.txt` to replace this:

```robots.txt
# Allow the latest release
#Disallow: /en/v0.8.3/
```

with this:
```robots.txt
# LATEST RELEASE: 0.8.3
```

Then, when version in `LATEST RELEASE` does not match, have the script add a new `Disallow` line before it and then update the version on the `LATEST RELEASE` line.

I'd also add some sanity checks:
- Exit with an error if a line of the form `# LATEST RELEASE: x.y.z` is not present or if there is more than one such line
- Exit with an error if there already is a `Disallow` directive that has the same version as the one used on the `LATEST RELEASE` line.

Reply:
I think that just looking at `Disallow` is not realiable enough. See for example how the current script badly updated `robots.txt` in the new PR.

I think that the system I suggested above would be more robust.